multiraft/multiraft.go

```diff
@@ -531,8 +625,16 @@ func (s *state) handleWriteResponse(response *writeResponse, readyGroups map[uin
 				}
 				log.V(3).Infof("node %v applying configuration change %v", s.nodeID, cc)
 				// TODO(bdarnell): dedupe by keeping a record of recently-applied commandIDs
-				// TODO(bdarnell): support removing nodes; fix double-application of initial entries
-				err = s.addNode(cc.NodeID)
+				switch cc.Type {
+				case raftpb.ConfChangeAddNode:
+					err = s.addNode(cc.NodeID, groupID)
+				case raftpb.ConfChangeRemoveNode:
+					// TODO(bdarnell): support removing nodes; fix double-application of initial entries
+					continue
+				case raftpb.ConfChangeUpdateNode:
+					// TODO(Tobias): What can be updated in such a request?
```

Comment:
@bdarnell looked for this in the raft paper and in the protobufs but no dice. Is there an explanation of what this does anywhere?